src/components/TodoappHeader/TodoappHeader.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { USER_ID, postTodo } from '../../api/todos';
+import { errorNotification } from '../../utils/errorFunction';
+
+interface TodoappHeaderProps {
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  todos: Todo[];
+  setErrorNotification: (msg: string) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+}
+
+export const TodoappHeader: React.FC<TodoappHeaderProps> = ({
+  setTodos,
+  todos,
+  setErrorNotification,
+  inputRef,
+}) => {
+  const [newTodo, setNewTodo] = useState<string>('');
+  const [activeTodo, setActiveTodo] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const everyActive = todos.length > 0 && todos.every(todo => todo.completed);
+
+    setActiveTodo(everyActive);
+  }, [todos]);
+
+  useEffect(() => {
+    if (!isLoading && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isLoading]);
```

Comment:
Fix it 
<img width="982" alt="image" src="https://github.com/user-attachments/assets/5fe293ad-6fbe-4dbe-8bdb-a8e76bda7fd6" />